core/src/test/java/test/org/apache/spark/JavaAPISuite.java

```diff
@@ -897,6 +901,16 @@ public void persist() {
     assertEquals(1, rdd.first().intValue());
   }
 
+  @Test
+  public void withResources() {
+    ExecutorResourceRequests ereqs = new ExecutorResourceRequests().cores(4);
+    TaskResourceRequests treqs = new TaskResourceRequests().cpus(1);
+    ResourceProfile rp1 = new ResourceProfileBuilder().require(ereqs).require(treqs).build();
+    JavaRDD<Integer> in1 = sc.parallelize(Arrays.asList(1, 2, 3, 4));
+    in1.withResources(rp1);
+    assertEquals(rp1, in1.getResourceProfile());
+  }
```

Comment:
Out of curiosity, what was the issue in the test?

Reply:
this was purely we don't support it in local mode. so before we were doing a in1.collect() which ended up hanging because we also have the isTesting config on.  it was looking for executor that it would never get.  Normally it would have thrown an exception saying local mode not supported but the isTesting flag bypassed that check